transport/TransportImpl.h

```diff
@@ -172,6 +172,7 @@ class TransportImpl : public RtcTransport,
     void runTick(uint64_t timestamp) override;
     ice::IceSession::State getIceState() const override { return _iceState; };
     SrtpClient::State getDtlsState() const override { return _dtlsState; };
+    utils::Optional<ice::TransportType> getSelectedTransportType() const override;
```

Comment:
I wonder if we should consider the option of having this in the DataReceiver callback interface. 
It feels like Transport is getting fatter and fatter with member variables and getters. 
We should perhaps give it a thought to evaluate the consequences. Perhaps this is the most light weight solution but we have more and more access methods to retrieve a little piece of the state. One option is to read out a struct over state using MpmcPublish, which then contains more, in synch, state information and is cheaper to read once from atomic than doing multiple atomic reads.

Reply:
I'd rather have it in a separate MR. What do you think?